src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -95,6 +96,12 @@ float64_t CrossValidation::evaluate_one_run(int64_t index) const
 
 	SGVector<float64_t> results(num_subsets);
 
+	std::vector<int32_t> seed_vector(num_subsets);
+	UniformIntDistribution<int32_t> seed_dist(std::numeric_limits<int32_t>::min(), 
+		std::numeric_limits<int32_t>::max());
```

Comment:
@vigsterkr not sure how else to generate int32_t values. Should seed be [0, 2^31-1] or [-2^31, 2^31-1]?

Reply:
why you just use the prng directly?

Reply:
because prng returns a uint64_t, which then would get truncated when casted to int32 no?

Reply:
why do you need signed int32_t? i mean it's gonna be a seed for another prng no?
for mersenne prng the seed should be actually unsigned 💃 so we have a bug in the Seedable :P

Reply:
So Seedable needs to know what the input type of the prng is?